app/views/agents/_table.html.erb

```diff
@@ -53,7 +53,7 @@
         </td>
         <td class='<%= "agent-unavailable" if agent.unavailable? %>'>
           <% if agent.can_create_events? %>
-            <%= link_to(agent.events_count || 0, agent_events_path(agent)) %>
+            <%= link_to(agent.events_count || 0, agent_events_path(agent, return: (defined?(return_to) && return_to) || request.path)) %>
```

Comment:
With fullpath it always returns you to Details page even if you came from scenario page